### ISSUE TITLE
Support FP16(amp) in imagenet main.py

### ISSUE THREAD

**Issue by Deeeerek** [pytorch/examples#1033]
I am running test under : https://github.com/pytorch/examples/tree/main/imagenet

Does the main.py support FP16(amp)?like arguments： --amp 

Thx

**Comment by hudeven** [comment]
@Deeeerek No. There was a try in https://github.com/pytorch/examples/pull/203 I think it should be easier and cleaner to support fp16 nowadays with amp from pytorch: https://pytorch.org/blog/accelerating-training-on-nvidia-gpus-with-pytorch-automatic-mixed-precision/ 
Please feel free to contribute if you like!

**Comment by Deeeerek** [comment]
Yes, I did it (amp) with `autocast, GradScaler`, much thx!
